source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -87,7 +87,11 @@ LoadClusterEntryHandlePtr ProxyFilterConfig::addDynamicCluster(
     ENVOY_LOG(debug, "deliver dynamic cluster {} creation to main thread", cluster_name);
     main_thread_dispatcher_.post([this, cluster, version_info]() {
       ENVOY_LOG(debug, "initializing dynamic cluster {} creation in main thread", cluster.name());
-      cluster_manager_.addOrUpdateCluster(cluster, version_info);
+
+      // Set ignore_removal to true to prevent the cluster from being removed during a CDS update.
+      // As this cluster lifecycle is managed by DFP cluster, it should not be removed by CDS.
+      // https://github.com/envoyproxy/envoy/issues/35171
+      cluster_manager_.addOrUpdateCluster(cluster, version_info, true);
```

Comment:
Makes sense that we don't want DFP sub-clusters removed by a CDS update, but just to check my understanding it is plausible that a non-sub-cluster DFP cluster be removed via CDS right? If so let's make sure we test that, and either way let's runtime guard this change.